parquet-column/src/main/java/org/apache/parquet/schema/LogicalTypeAnnotation.java

```diff
@@ -77,63 +77,116 @@ static LogicalTypeAnnotation fromOriginalType(OriginalType originalType, Decimal
       case UINT_32:
-        return IntLogicalTypeAnnotation.create(32, false);
+        return intType(32, false);
       case UINT_64:
-        return IntLogicalTypeAnnotation.create(64, false);
+        return intType(64, false);
       case INT_8:
-        return IntLogicalTypeAnnotation.create(8, true);
+        return intType(8, true);
       case INT_16:
-        return IntLogicalTypeAnnotation.create(16, true);
+        return intType(16, true);
       case INT_32:
-        return IntLogicalTypeAnnotation.create(32, true);
+        return intType(32, true);
       case INT_64:
-        return IntLogicalTypeAnnotation.create(64, true);
+        return intType(64, true);
       case ENUM:
-        return EnumLogicalTypeAnnotation.create();
+        return enumType();
       case JSON:
-        return JsonLogicalTypeAnnotation.create();
+        return jsonType();
       case BSON:
-        return BsonLogicalTypeAnnotation.create();
+        return bsonType();
       case MAP_KEY_VALUE:
-        return MapKeyValueTypeAnnotation.create();
+        return mapKeyValueType();
       default:
         throw new RuntimeException("Can't convert original type to logical type, unknown original type " + originalType);
     }
   }
 
+
+  static StringLogicalTypeAnnotation stringType() {
+    return StringLogicalTypeAnnotation.INSTANCE;
+  }
+
+  static MapLogicalTypeAnnotation mapType() {
+    return MapLogicalTypeAnnotation.INSTANCE;
+  }
+
+  static ListLogicalTypeAnnotation listType() {
+    return ListLogicalTypeAnnotation.INSTANCE;
+  }
+
+  static EnumLogicalTypeAnnotation enumType() {
+    return EnumLogicalTypeAnnotation.INSTANCE;
+  }
+
+  static DecimalLogicalTypeAnnotation decimalType(final int scale, final int precision) {
+    return new DecimalLogicalTypeAnnotation(scale, precision);
+  }
+
+  static DateLogicalTypeAnnotation dateType() {
+    return DateLogicalTypeAnnotation.INSTANCE;
+  }
+
+  static TimeLogicalTypeAnnotation timeType(final boolean isAdjustedToUTC, final TimeUnit unit) {
+    return new TimeLogicalTypeAnnotation(isAdjustedToUTC, unit);
+  }
+
+  static TimestampLogicalTypeAnnotation timestampType(final boolean isAdjustedToUTC, final TimeUnit unit) {
+    return new TimestampLogicalTypeAnnotation(isAdjustedToUTC, unit);
+  }
+
+  static IntLogicalTypeAnnotation intType(final int bitWidth, final boolean isSigned) {
+    Preconditions.checkArgument(
+      bitWidth == 8 || bitWidth == 16 || bitWidth == 32 || bitWidth == 64,
+      "Invalid bit width for integer logical type, " + bitWidth + " is not allowed, " +
+        "valid bit width values: 8, 16, 32, 64");
+    return new IntLogicalTypeAnnotation(bitWidth, isSigned);
+  }
+
+  static JsonLogicalTypeAnnotation jsonType() {
+    return JsonLogicalTypeAnnotation.INSTANCE;
+  }
+
+  static BsonLogicalTypeAnnotation bsonType() {
+    return BsonLogicalTypeAnnotation.INSTANCE;
+  }
+
+  static IntervalLogicalTypeAnnotation intervalType() {
```

Comment:
Do we want the user to be able to create an Interval or a MapKeyValue type? If they are only exist for backward compatibility reasons the factory method should not be public so users won't be able to create them.

Reply:
I think MapKeyValue is here only for backward compatibility, but Interval should be supported in the future. Since these factory methods are defined on an interface, I can't easily change the visibility to private/package protected. Should I change it to an abstract class?

Reply:
Interval is not supported in LogicalType currently, so I would not expose it yet. It is always possible to make a restricted access public but in the other way it would break backward compatibility.
Abstract class sounds reasonable to me.